src/main/kotlin/no/skatteetaten/aurora/databasehotel/service/DatabaseHotelService.kt

```diff
@@ -116,11 +117,15 @@ class DatabaseHotelService(private val databaseHotelAdminService: DatabaseHotelA
         } ?: throw IllegalArgumentException("no database schema found for id: $id")
 
     fun validateConnection(jdbcUrl: String, username: String, password: String) =
-        try {
-            DriverManager.getConnection(jdbcUrl, username, password).use { true }
-        } catch (ex: SQLException) {
-            false
-        }
+            try {
+                DriverManager.getConnection(jdbcUrl, username, password).use {
+                    ConnectionVerificationResponse(true, "success")
+                }
+
+            } catch (ex: SQLException) {
+                logger.info { ex.message }
+                ConnectionVerificationResponse(false, ex.message)
```

Comment:
Det er viktig at service-laget ikke bruker klasser fra api/web-laget. Nå har du i praksis introdusert en sirkulær avhengighet mellom de to lagene (som ikke er bra). Du kunne flyttet ConnectionVerification til servicelaget i nærheten av validateConnection-metoden, slik at det blir tydelig at validateConnection-metoden "eier" retur-typen sin, og dernest returnert dette objektet direkte fra web-laget. Men jeg vil faktisk anbefale deg å lage to dataklasser, selv om de i dette tilfellet blir like. På den måten har du frihet til å utforme respons fra servicelaget og respons fra weblaget helt individuelt. Det får du i praksis brukt for når du må støtte både v1 og v2 av dette endepunktet (for å unngå at gobo brekker). PS Kanskje vi bør høre med Bjarte om hvordan vi skal håndtere endringen i responsen på dette endepunktet?

Reply:
To dataklasser laget